xarray/backends/api.py

```diff
@@ -1578,7 +1622,7 @@ def to_zarr(
     compute: bool = True,
     consolidated: bool | None = None,
     append_dim: Hashable | None = None,
-    region: Mapping[str, slice] | None = None,
+    region: Mapping[str, slice | Literal["auto"]] | None = None,
```

Comment:
We could also offer `region="auto"` to run the `"auto"` on all dims (but not required)

Reply:
Oh yeah, that's an obvious improvement, thanks. Now I'm wondering, is there any reason not to have this be the only auto option, i.e. `Mapping[str, slice] | Literal["auto"]`? Would there ever be a need to do a mix e.g. `region={"x": "auto", "y": slice(0, 10)}`?

Reply:
Yes good point, I think that's a simpler API.

I guess there's possibly some corner cases where we have one index but not the other. No strong view on whether it's worth the additional complexity to enable those...